test/ng/compileSpec.js

```diff
@@ -980,6 +980,29 @@ describe('$compile', function() {
         }));
 
 
+        it('should not set merged attributes twice in $attrs', function() {
+          var attrs;
+
+          module(function() {
+            directive('logAttrs', function() {
+              return {
+                link: function($scope, $element, $attrs) {
+                  attrs = $attrs;
+                }
+              }
+            })
+          });
+
+          inject(function($compile, $rootScope) {
+            element = $compile(
+              '<div><div log-attrs replace class="myLog"></div><div>')($rootScope);
+            var div = element.find('div');
+            expect(div.attr('class')).toBe('myLog log');
+            expect(attrs.class).toBe('myLog log');
+          });
+        });
```

Comment:
There are different `if/else` branhes for each. Most importantly, the one uses `safeAddClass(...)` and the other `$element.attr('style', ...)`. We need to make sure that whatever these code paths do (which are skipped when `dst[key]` is trythy, has already been done in the previous loop (where we use `dst.$set(...)`).